src/map.cpp

```diff
@@ -3496,7 +3526,8 @@ bool map::open_door( const tripoint &p, const bool inside, const bool check_only
         }
 
         if(!check_only) {
-            sounds::sound(p, 6, "", true, "open_door", furn.id);
+            sounds::sound(p, 6, "", true, "", "");
+            sfx::play_variant_sound( "open_door", ter.id, sfx::get_heard_volume(p));
```

Comment:
Considering how often you write `get_heard_volume` in the call to `play_variant_sound` (like here), there should probably be an overload for `play_variant_sound`, which accepts a `tripoint` instead of the heard volume and calculates the volume itself:
```C++
void play_variant_sound( const std::string &a, const std::string &b, const tripoint &p ) {
    play_variant_sound( a, b, get_heard_volume( p ) );
}
```

Anyway, this PR has huge potential.